packages/babel-helpers/src/helpers.js

```diff
@@ -434,16 +434,23 @@ helpers.construct = () => template.program.ast`
   import setPrototypeOf from "setPrototypeOf";
 
   export default function _construct(Parent, args, Class) {
-    if (typeof Reflect !== "undefined" && Reflect.construct) {
-      _construct = Reflect.construct;
+    if (
+      typeof Reflect !== "undefined" && Reflect.construct
+    ) {
+      _construct = function _construct(Parent, args, Class) {
+        // This wrapper is needed because Reflect.construct can't be properly
+        // polyfilled, thus core-js doesn't set the correct __proto__.
```

Comment:
This fails to fix `Date` construction when Reflect.construct is polyfilled. We have to test for sham implementations, fall back to the bind `_construct` below when it's a sham.

https://babeljs.io/repl/build/7946/#?babili=false&browsers=&build=&builtIns=false&code_lz=PTAEBUAsEsGdTqAprWSB2AXaBDANngJ6gDGA9gE5IC0AVrAOSgBKSAZnkiZgHTnqxMFAK7dQABzJE20AgChWHLr36CRYgLyg2w9N2hl0ACnA4KAcySYANKDPnYt9EgDupi1YCUoAN5zQAaAAbmYSFGSYZKBazm72Vjzi4ZGYhOJIANz-gSEUCAKYOHpI0aAA8gBGtMp8VDiYSEZJEWSeWYHBoVSwwniYpQBiuvqGPDji4kR8-Hgm8Tb5gkUkSLb2sG3ZAVSYwhTooN29_QBkJ6Cp6WRshyjH0RpaDGRVykwA_Lc9faAAXIuFYpZAC-WTkJDwOFQoAAsoQACL1EpIAAeDXQABN4IiGr5sqohKJIhQjNgALZIbx-DoBHrpEnkyntUDAuSszj9MkIpGlWKw7kNIybcGGWBSJBjVBICiYIxcnESyIAZSE0HQ5geWgViWSETSirIKooavMm1U4slaBlpP111A8qRPGVqvVQs1oAARGoTR62kA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=true&fileSize=false&lineWrap=true&presets=es2015&prettier=false&targets=&version=7.0.0-beta.46%2Bpr.7533&envVersion=

Reply:
Is there another way other than try/catch?

Reply:
[Something like this](https://babeljs.io/repl/build/7946/#?babili=false&browsers=&build=&builtIns=false&code_lz=PTAEBUAsEsGdTqAprWSB2AXaBDANngJ6gDGA9gE5IC0AVrAOSgBKSAZnkiZgHTnqxMFAK7dQABzJE20AgChWHLr36CRYgLyg2w9N2hl0ACnA4KAcySYANKDPnYt9EgDupi1YCUoAN5zQAaAAbmYSFGSYZKBazm72Vjzi4ZGYhOJIANz-gSEUCAKYOHpI0aAA8gBGtMp8VDiYSEZJEWSeWYHBoVSwwniYpQBiuvqGPDji4kR8-Hgm8Tb5gkUkSLb2sG3ZAVSYwhTooN29_QBkJ6Cp6WRshyjH0RpaDGRVykwA_Lc9faAAXIuFYpZAC-WTknH6sEgOAAtqU2Pg0FkhMQ_AEACL1JCJZIRNLYyIAZSE0HQ5mmBCMWxY7E43D4hjUokwRkxDVsAG0ALq2HR6bCGIzeHzAzzZTbA0j1EiQUBGJDC7JQ2GlITCTJyYFyVRSbF4MjmIzKmFtIA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=true&fileSize=false&lineWrap=true&presets=es2015&prettier=false&targets=&version=7.0.0-beta.46%2Bpr.7533&envVersion=) is the only foolproof way to detect if they're not using a `Binder` polyfill like we are.

core-js@3 added a `sham` property to the method, but that won't help anyone using an older version.